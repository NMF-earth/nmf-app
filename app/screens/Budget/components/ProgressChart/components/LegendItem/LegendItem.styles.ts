import { StyleSheet } from "react-native";

import { Font } from "style";

export default StyleSheet.create({
  container: {
    justifyContent: "center",
    flexDirection: "row",
    paddingTop: 10,
  },
  coloredDot: {
    width: Font.FontSize.Secondary + 2,
    height: Font.FontSize.Secondary + 2,
    borderRadius: 10,
  },
  coloredDotView: {
    alignSelf: "center",
    marginHorizontal: 6,
    width: Font.FontSize.Secondary,
    height: "100%",
  },
});
