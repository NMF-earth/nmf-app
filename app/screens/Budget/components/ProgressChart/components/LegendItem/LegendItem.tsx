import React from "react";
import { View } from "react-native";

import styles from "./LegendItem.styles";
import { Text } from "../../../../../../components";

interface Props {
  name: string;
  value: number;
  color: string;
}

const LegendItem = ({ name = "", value = 0, color = "" }: Props) => {
  if (value === 0) {
    return null;
  }

  return (
    <View style={styles.container}>
      <View style={styles.coloredDotView}>
        <View style={[{ backgroundColor: color }, styles.coloredDot]} />
      </View>
      <View>
        <Text.Secondary bold>{name}</Text.Secondary>
        <Text.Secondary lightGray>{value + " kg"}</Text.Secondary>
      </View>
    </View>
  );
};

export default LegendItem;
