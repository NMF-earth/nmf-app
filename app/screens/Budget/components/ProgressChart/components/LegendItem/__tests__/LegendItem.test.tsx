import React from "react";
import { create } from "react-test-renderer";

import LegendItem from "../LegendItem";

const props = {
  name: "Food",
  value: 10,
  color: "red",
};

jest.unmock("../LegendItem");

it("LegendItem renders correctly", () => {
  const tree = create(<LegendItem {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("LegendItem should not render if value is 0", () => {
  const tree = create(<LegendItem {...props} value={0} />).toJSON();
  expect(tree).toBeNull();
});
