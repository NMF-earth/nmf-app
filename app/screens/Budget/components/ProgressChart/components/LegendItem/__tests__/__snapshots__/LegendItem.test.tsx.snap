// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendItem renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingTop": 10,
    }
  }
>
  <Text.Secondary>
    Food : 
  </Text.Secondary>
  <Text.Secondary
    lightGray={true}
  >
    <FormattedNumber
      maximumFractionDigits={1}
      value={60}
    />
     
     kg - 
  </Text.Secondary>
  <Text.Secondary
    lightGray={true}
  >
    60 %
  </Text.Secondary>
</View>
`;

exports[`LegendItem should not render % if percentage is 0 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingTop": 10,
    }
  }
>
  <Text.Secondary>
    Food : 
  </Text.Secondary>
  <Text.Secondary
    lightGray={true}
  >
    <FormattedNumber
      maximumFractionDigits={1}
      value={10}
    />
     
     g
  </Text.Secondary>
</View>
`;
